docs/runtimes/create-and-use-functions.ipynb

```diff
@@ -0,0 +1,550 @@
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "89e8a5ad",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "fn = project.set_function(name=\"describe\", func=\"hub://describe\")\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fd2d5a82",
+   "metadata": {},
+   "source": [
+    "#### MLRun DB"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "66397b78",
+   "metadata": {},
+   "source": [
+    "You can also import functions directly from the MLRun DB. These might be functions that have not been pushed to a git repo, archive, or function marketplace. Import via the name of the function and the `db://` prefix:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a69e3f7f",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "fn = project.set_function(name=\"db\", func=\"db://import\")\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "58c92849",
+   "metadata": {},
+   "source": [
+    "#### MLRun Function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "69c0d1c5",
+   "metadata": {},
+   "source": [
+    "Finally, you can also directly use an existing MLRun function object. This is usually used when more granular control over function parameters is required (e.g. advanced parameters not supported by {py:meth}`~mlrun.projects.MlrunProject.set_function`)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7ceb2812",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "fn = mlrun.new_function(\"serving\", kind=\"serving\", image=\"mlrun/mlrun\")\n",
+    "graph = serving.set_topology(\"flow\")\n",
+    "graph.to(name=\"double\", handler=\"mylib.double\") \\\n",
+    "     .to(name=\"add3\", handler=\"mylib.add3\") \\\n",
+    "     .to(name=\"echo\", handler=\"mylib.echo\").respond()\n",
+    "\n",
+    "project.set_function(name=\"serving\", func=fn, with_repo=True)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0b2601c1",
+   "metadata": {},
+   "source": [
+    "<a id=\"customizing-functions\"></a>\n",
+    "## Customizing Functions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6bed5e82",
+   "metadata": {},
+   "source": [
+    "Once you have created your MLRun function, there are many customizations you can add. Some potential customizations include:\n",
+    "\n",
+    "#### Environment Variables\n",
+    "Environment variables can be added individually, from a Python dictionary, or a file\n",
+    "\n",
+    "```python\n",
+    "# Single variable\n",
+    "fn.set_env(name=\"MY_ENV\", value=\"MY_VAL\")\n",
+    "\n",
+    "# Multiple variables\n",
+    "fn.set_envs(env_vars={\"MY_ENV\" : \"MY_VAL\", \"SECOND_ENV\" : \"SECOND_VAL\"})\n",
+    "\n",
+    "# Multiple variables from file\n",
+    "fn.set_envs(file_path=\"env.txt\")\n",
+    "```\n",
+    "\n",
+    "#### Memory, CPU, GPU Resources\n",
```

Comment:
"#### Memory, CPU, GPU resources\n",
    "Adding requests and limits to your function specifies the required compute resources. It is best practice to define this for each MLRun function. See [**CPU, GPU, and memory limits for user jobs**](./configuring-job-resources.html#cpu-gpu-and-memory-limits-for-user-jobs) for more information on configuring resources.\n",